sycl/include/sycl/ext/intel/experimental/esimd/detail/memory_intrin.hpp

```diff
@@ -456,8 +466,9 @@ __esimd_scatter_scaled(__SEIEED::simd_mask_storage_t<N> pred,
 
     for (int idx = 0; idx < N; idx++) {
       if (pred[idx]) {
-        Ty *addr = reinterpret_cast<Ty *>(elem_offsets[idx] + writeBase);
-        *addr = vals[idx];
+        ActualTy *addr =
```

Comment:
ditto

Reply:
how this compiles, btw?

Reply:
No problem in compilation. But, some floating point tests are failing with mismatches after applying your suggestions as far as I can understand.